### PR TITLE
#2110 Refactor MessageGroupMetadata to be compatible with Jackson serialization

### PR DESCRIPTION
**Pull request by urgelacko**
Connected to https://github.com/spring-projects/spring-integration/issues/2110

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@urgelacko Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2112)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2112) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@urgelacko Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=2112)!

File: spring-integration-core/src/test/java/org/springframework/integration/store/MessageGroupMetadataJacksonTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Laszlo Szabo
+ */
+public class MessageGroupMetadataJacksonTest {
```
Comment (artembilan) [inline spring-integration-core/src/test/java/org/springframework/integration/store/MessageGroupMetadataJacksonTest.java]: OK, @urgelacko ,

that's great contribution!

Would you mind to add JSON-based test for the `RedisMessageStore`?
As far as that one was the reason to come here with the concerns

Thank you!

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-integration-core/src/test/java/org/springframework/integration/store/MessageGroupMetadataJacksonTest.java (hunk 1 above)
Comment (urgelacko) [inline spring-integration-core/src/test/java/org/springframework/integration/store/MessageGroupMetadataJacksonTest.java]: Yes, I will try to add it soon.
Thanks!

**Comment by artembilan** [comment]
@urgelacko ,

I have reviewed this ad realized that the situation is much complicated than just `MessageGroupMetadata`.
I've gone ahead and implemented this #2138 .

I used your fix for the `MessageGroupMetadata` and followed with similar way in some other places.
Since that PR covers this change as well, we are closing this one.

Your feedback in that PR is much than welcome.

And thank you for the contribution!

**Comment by urgelacko** [comment]
@artembilan 
Sorry for disappearing but recently I had zero time to implement this. Thank you for finishing and adding this to SI.
Bests

**Pull request closed without merging**